frontend/src/components/ProgressBar.tsx

```diff
@@ -0,0 +1,44 @@
+// ProgressBar.js
+import { FC } from 'react';
+
+interface ProgressBarProps {
+    progress: number;
+}
+
+const ProgressBar: FC<ProgressBarProps> = ({ progress }) => {
+  return (
+    <div className="w-full bg-gray-300 rounded-full relative">
+      <div
+        className="
+          flex
+          items-center
+          justify-center
+          bg-indigo-600
+          rounded-full
+          transition-width
+          duration-300
+          ease-in-out"
+        style={{ width: `${progress}%`, height: '8px' }}
+      />
+      <div
+        className="
+          bg-indigo-600
+          rounded-full
+          absolute
+          transform
+          translate-x-1/2
+          transition-width
+          duration-300
+          ease-in-out"
+        style={{
+          width: '12px',
+          height: '12px',
+          top: '-2px',
+          left: `${Math.max(progress - 2, -1)}%`
```

Comment:
the circle needs to be a bit more to the left to make it congruent with figma design
```suggestion
          left: `${Math.max(progress - 3, -1)}%`
```

Reply:
<img width="412" alt="image" src="https://github.com/devsoc-unsw/trainee-uranus-24t1/assets/46995931/dd139be2-f177-43a8-b014-a8c04c534e46">

<img width="344" alt="image" src="https://github.com/devsoc-unsw/trainee-uranus-24t1/assets/46995931/b95fcbf5-c818-4f0a-a94d-4c568143f8b7">